textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -36,5 +50,27 @@ public boolean satisfy(ITuple tuple) {
         }
         return false;
     }
+    
+    public List<Span> statisfySpan(ITuple tuple) {
+    	List<Span> res = new ArrayList<>();
+    	if (tuple == null) {
+    		return res; //empty array
+    	}
+    	IField field = tuple.getField(fieldName);

```

Comment:
@chenlica .. Yes we will support both `StringField` and `TextField`. If we are searching in a `StringField` then the entire key should be matched with that field. Whereas `TextField` supports partial matching.